File: LeetCode/medium/Add_Two_Numbers.cxx
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        int r = 0;
        ListNode *nouv = NULL;
        ListNode *res = nouv;
        while (l1 || l2 || r > 0) {
            int value = r;
            if (l1)
            {
                value += (l1)->val;
                l1 = l1->next;                
            }
            if (l2)
            {
                value += (l2)->val;
                l2 = l2->next;                
            }
            if (!nouv)
            {
                nouv = new ListNode(0);
                res = nouv;
            } else
            {
                nouv->next = new ListNode(0);
                nouv = nouv->next;
            }
            
            nouv->val = value % 10;
            r = value / 10;
        }
        return res;
    }
};

File: LeetCode/medium/Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        retenue = 0
        a = l1
        b = l2
        res = ListNode(-1)
        cur = res
        while a is not None or b is not None:
            s = retenue
            if a is not None:
                s += a.val
                a = a.next
            if b is not None:
                s += b.val
                b = b.next
            cur.next = ListNode(s % 10)
            cur = cur.next
            retenue = s // 10
        while retenue > 0:
            cur.next = ListNode(retenue % 10)
            retenue = retenue // 10
        return res.next
            
